pkg/infra/game/agent/agent.go

```diff
@@ -12,39 +12,49 @@ import (
 	"github.com/benbjohnson/immutable"
 )
 
-type Strategy interface {
-	HandleFightMessage(m message.TaggedMessage, view *state.View, agent BaseAgent, log *immutable.Map[commons.ID, decision.FightAction]) decision.FightAction
-	Default() decision.FightAction
-}
-
 type Agent struct {
 	BaseAgent BaseAgent
-	Strategy  Strategy
+	Strategy  message.Strategy
```

Comment:
Is there any particular reason as to moving Strategy to message? I can sorta understand it since the strategy is tied to message handling?

Reply:
Yeah basically because strategy references messages and messages references strategy I was trying to avoid cyclic imports

Reply:
I assume theres probs a way to seperate it out though